### PR TITLE
Add `Likes` button to the navbar

### PR DESCRIPTION
**Pull request by vcapretz**
when we want to access our liked tweets the current flow is:
click profile picture > profile > find likes button

with this new button you can click whenever you are at

screen shot 2018-02-08 at 15 34 59

### CODE REVIEW

File: source/content.js [hunk 1]
```diff
@@ -48,6 +59,7 @@ function init() {
 	onRouteChange(() => {
 		safely(autoLoadNewTweets);
 		safely(userChoiceColor);
+		addLikeButtonNavbar();
```
Comment (sindresorhus) [inline source/content.js]: Use `safely()` like the other calls above.

File: source/content.js [hunk 2]
```diff
@@ -25,6 +25,17 @@ function onNewTweets(cb) {
...
 
+function addLikeButtonNavbar() {
+	$('#global-actions').append(`
+		<li>
+			<a role="button" data-nav="favorites" href="/i/likes" class="js-nav js-tooltip js-dynamic-tooltip" data-placement="bottom" data-original-title="">
+				<span class="Icon Icon--heart Icon--large"></span>
+				<span class="text">Likes</span>
+			</a>
+		</li>
+	`);
```
Comment (sindresorhus) [inline source/content.js]: Use JSX. Example: https://github.com/sindresorhus/refined-twitter/blob/ae0bc9985fda2fd09f26b30915163c666e9eaf46/source/features/code-highlight.js#L64-L76

**Comment by sindresorhus** [comment]
The `Likes` button is added quite slowly. Many hundred milliseconds after the page has loaded.

And if I press the `Likes` button, it's added again. You need to check and prevent duplicate buttons. Example: https://github.com/sindresorhus/refined-github/blob/0d25d590e828fb54db2ad8342f8115aa4aa23ec8/source/features/add-gists-link-to-profile.js#L7-L11

File: source/content.js [hunk 3]
```diff
@@ -25,6 +25,17 @@ function onNewTweets(cb) {
 	observeEl('#stream-items-id', cb);
 }
 
+function addLikeButtonNavbar() {
+	$('#global-actions').append(`
+		<li>
+			<a role="button" data-nav="favorites" href="/i/likes" class="js-nav js-tooltip js-dynamic-tooltip" data-placement="bottom" data-original-title="">
```
Comment (sindresorhus) [inline source/content.js]: I don't think you need to add `data-original-title=""`. It's added by the JS on hover.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by vcapretz** [comment]
@sindresorhus thanks for the review! I just fixed them.

let me know if there's still something to be done 😁

File: source/features/likes-button-navbar.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import {h} from 'dom-chef';
+
+export default function () {
```
Comment (sindresorhus) [inline source/features/likes-button-navbar.js]: Use arrow function

File: source/features/likes-button-navbar.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+	const globalActionsList = $('#global-actions');
+
+	if (globalActionsList.find('.refined-like-button').length > 0) {
```
Comment (sindresorhus) [inline source/features/likes-button-navbar.js]: `.refined-like-button` => `.refined-twitter_like-button`

**Comment by sindresorhus** [comment]
It still loads pretty slowly. It's not a good experience when the button is visibly loaded after page load. @bfred-it Any tips for making it appear before the page is painted?

**Comment by fregante** [comment]
Don’t wait for domready, use safeElementReady (this means you have to restore `run_at: document_start`)

**Comment by sindresorhus** [comment]
@bfred-it Cool. I reverted your commit then: https://github.com/sindresorhus/refined-twitter/commit/fe35d0a98162a5c5aa7a0c6bedf5aa9d81192d01

**Comment by sindresorhus** [comment]
@vcapretz Can you use `safeElementReady`? Example here: https://github.com/sindresorhus/refined-twitter/blob/fe35d0a98162a5c5aa7a0c6bedf5aa9d81192d01/source/content.js#L22

**Comment by sindresorhus** [comment]
@vcapretz Can you also add this feature to the feature listing in the readme?

**Review by fregante — APPROVED** [review]

**Review by filipekiss — APPROVED** [review]

**Comment by vcapretz** [comment]
sure @sindresorhus, I'll update it

**Comment by vcapretz** [comment]
hey, @sindresorhus I just applied the suggested changes! 

now the function runs on init, and I saw it has really improved the loading performance.

I also added the `safeElementReady` function to the main list where the new item is about to be added!

**Review by sindresorhus — APPROVED** [review]

**Comment by sindresorhus** [comment]
Great. Thanks :)

**Pull request merged**